spec/dummy/spec/packs_generator_spec.rb

```diff
@@ -259,6 +260,28 @@ def create_new_component(name)
       end
     end
 
+    context "when components subdirectory is not set" do
+      before do
+        ReactOnRails.configure do |config|
+          config.components_subdirectory = nil
+          config.auto_load_bundle = false
+        end
+      end
```

Comment:
I am still trying to figure out this part.
Does it mean if one sets `components_subdirectory` to `nil`, they should set `auto_load_bundle` also to false? If so, we should document it. If not, we should automate it.

Otherwise, we may need to implement tests differently.

I am waiting for updates.

Reply:
@ahangarha was trying different things, but can't quite figure out why that suite is failing. (Also having a hard time running all the tests locally).

There is already some check that throws an error if `components_subdirectory` is `nil` and `auto_load_bundle` is `true` (in `Configuration#check_autobundling_requirements_if_configured`).

Reply:
Thanks for your work so far.

So, while you are working on the issue, I also try to fix the problem. Please update the PR if you made any progress.

Reply:
@ahangarha if `auto_load_bundle` is true & `components_subdirectory` is not set, then [the Configuration throws an error here](https://github.com/shakacode/react_on_rails/blob/16a4c7d16f6072e682ce91398c03c22500bc21d4/lib/react_on_rails/configuration.rb#L128)

This PR is to stop a bug, in which [the PacksGenerator runs during the `assets:precompile` rake task](https://github.com/shakacode/react_on_rails/blob/16a4c7d16f6072e682ce91398c03c22500bc21d4/lib/react_on_rails/configuration.rb#L160), even though `auto_load_bundle` is `false` & `components_subdirectory` is not set.